scripts/globals/pets/wyvern.lua

```diff
@@ -116,13 +116,25 @@ function onMobSpawn(mob)
                 weaknessTargetChance = 100
             end
             if math.random(100) <= weaknessTargetChance then
-                local weakness = 0
-                for mod = 0, 5 do
-                    if target:getMod(tpz.mod.FIREDEF + mod) < target:getMod(tpz.mod.FIREDEF + weakness) then
-                        breaths = {}
-                        table.insert(breaths, tpz.jobAbility.FLAME_BREATH + mod)
-                    elseif target:getMod(tpz.mod.FIREDEF + mod) == target:getMod(tpz.mod.FIREDEF + weakness) then
-                        table.insert(breaths, tpz.jobAbility.FLAME_BREATH + mod)
+                local resistances =
+                {
+                    [tpz.jobAbility.FLAME_BREATH] = target:getMod(tpz.mod.FIRERES),
+                    [tpz.jobAbility.FROST_BREATH] = target:getMod(tpz.mod.ICERES),
+                    [tpz.jobAbility.GUST_BREATH] = target:getMod(tpz.mod.WINDRES),
+                    [tpz.jobAbility.SAND_BREATH] = target:getMod(tpz.mod.EARTHRES),
+                    [tpz.jobAbility.LIGHTNING_BREATH] = target:getMod(tpz.mod.THUNDERRES),
+                    [tpz.jobAbility.HYDRO_BREATH] = target:getMod(tpz.mod.WATERRES)
+                }
+                local lowest = resistances[tpz.jobAbility.FLAME_BREATH]
+                for key, value in pairs(resistances) do
+                    if value < lowest then
+                        lowest = value
+                    end
+                end
+                for key, value in pairs(resistances) do
+                    if value == lowest then
+                        table.insert(breaths, key)
+                        break
```

Comment:
You can cut the second loop by assigning your `breath` as you're stepping through the table the first time. This will be guaranteed to only be a single breath when you exit the loop because your resistance comparisons are strictly less than and don't include equal to. Then you can `table.insert(breaths, breath)` directly.

Reply:
In making this change I noticed that I needed to convert to using ipairs, as the table order was arbitrary using pairs. Corrected this at the same time!